src/azure-cli/azure/cli/command_modules/synapse/operations/workspace.py

```diff
@@ -29,14 +37,23 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, storage_a
         sql_administrator_login_password=sql_admin_login_password,
         location=location,
         managed_virtual_network="default" if enable_managed_virtual_network is True else None,
+        managed_virtual_network_settings=managed_virtual_network_settings,
+        encryption=encryption,
         tags=tags
     )
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, workspace_name, workspace_info)
 
 
 def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin_login_password=None,
-                     tags=None, no_wait=False):
-    workspace_patch_info = WorkspacePatchInfo(tags=tags, sql_admin_login_password=sql_admin_login_password)
+                     allowed_aad_tenant_ids=None, disable_all_allowed_aad_tenant_ids=None, tags=None, key_name=None, key_identifier=None, no_wait=False):
```

Comment:
>allowed_aad_tenant_ids=None, disable_all_allowed_aad_tenant_ids=None, [](start = 21, length = 69)

Is it possible to combine these two parameters? For example, is it possible for users to pass an empty list (v.s. None value) to CLI?

Reply:
allowed_aad_tenant_ids  is a List type defined in CLI, so it can't be Empty when user add it as parameter. 

Can we design it like "az synapse workspace update --allowed_aad_tenant_ids  None", if allowed_aad_tenant_ids  list contains "None", will disable all tenant ids.

```
    if allowed_aad_tenant_ids is not None and "None" in allowed_aad_tenant_ids:
        allowed_aad_tenant_ids = []
```

Reply:
@jsntcy , could you please provide some insights on this? What is the general guidelines for users to update a list to none or empty.

Reply:
I think you should use another command to clear the list, for example `az synapse workspace disable_allowed_aad_tenant_ids` to do that.

Reply:
Got it. Will change it back using `disable_allowed_aad_tenant_ids`  parameter. Thanks @idear1203  @jsntcy

Reply:
Find a similar cmdlet: 
```
c.argument('key_ops', arg_type=get_enum_type(JsonWebKeyOperation), options_list=['--ops'], nargs='*',
```
`nargs='*'` instead of `nargs='+' `, user can pass an empty list 


https://github.com/Azure/azure-cli/blob/e015d5bcba0c2d21dc42189daa43dc1eb82d2485/src/azure-cli/azure/cli/command_modules/keyvault/_params.py#L298

Reply:
Refer to https://github.com/Azure/azure-cli/blob/5aa3d4e45f4894efe48a6b2606c32993e21cef96/src/azure-cli/azure/cli/command_modules/network/_params.py#L1106
@sunsw1994